Consistency fixes (#60)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

Some Transparency Services may also enforce role based access control (RBAC) policies limiting who can write, read and audit specific Feeds or the full Registry.
It is critical to provide interoperability for all Transparency Services instances as the composition and configuration of involved supply chain entities and their system components is ever-changing and always in flux.

A Transparency Services provides visibility into Signed Statements associated with various supply chains and their sub-systems.
These Signed Statements (and corresponding Statement payload) are about the Artifacts produced by a supply chain.
A Transparency Service endorses specific and well-defined metadata about these Artifacts that is captured in Statements.
Some metadata is selected (and signed) by the Issuer, indicating, e.g., "who issued the Statement" or "what type of Artifact is described" or "what is the Artifact's version"; whereas additional metadata is selected (and countersigned) by the Transparency Services, indicating, e.g., "when was the Signed Statement about the Artifact registered in the Registry".

> The countersignature is often referred to as being notarized.

Reviewer: ```suggestion
Producing a Transparent Statement is also referred to as a form of notarization.
```

Author: Added. Note, I'm not using the GithUb commit suggestion as it bypasses signing :(

Producers of Signed Statement rely on the Transparency Service being discoverable and represented as the responsible parties for their Registered Signed Statements.
Analogously, Transparent Statement Consumers rely on verifiable trustworthiness assertions associated with Transparent Statements and their processing provenance.
If trust can be put into the operations that record Signed Statements (i.e., a verified notarization function) in a secure, append-only Ledger via online operations, the same trust can be put into a corresponding Receipt that is the resulting documentation of these online operations issued by the Transparency Services and that can be validated in offline operations.

The Transparency Services specified in this architecture can be implemented by various different types of services in various types of languages provided via various variants of API layouts.

The interoperability guaranteed by the Transparency Services is enabled via core components (architectural constituents) that come with prescriptive requirements (that are typically hidden away from the user audience via APIs).

> The core components are based on the Concise Signing and Encryption standard specified in {{-COSE}}, which is used to Sign Statements about Artifacts and to build and maintain a Merkle tree that functions as an append-only Ledger for corresponding Signed Statements.

Reviewer: ```suggestion
The core components are based on the Concise Signing and Encryption standard specified in {{-COSE}}, which is used to produce Signed Statements about Artifacts and to build and maintain a Merkle tree that functions as an append-only Ledger for corresponding Signed Statements.
```

Some Transparency Services may also enforce role based access control (RBAC) policies limiting who can write, read and audit specific Feeds or the full Registry.
It is critical to provide interoperability for all Transparency Services instances as the composition and configuration of involved supply chain entities and their system components is ever-changing and always in flux.

A Transparency Services provides visibility into Signed Statements associated with various supply chains and their sub-systems.
These Signed Statements (and corresponding Statement payload) are about the Artifacts produced by a supply chain.
A Transparency Service endorses specific and well-defined metadata about these Artifacts that is captured in Statements.
Some metadata is selected (and signed) by the Issuer, indicating, e.g., "who issued the Statement" or "what type of Artifact is described" or "what is the Artifact's version"; whereas additional metadata is selected (and countersigned) by the Transparency Services, indicating, e.g., "when was the Signed Statement about the Artifact registered in the Registry".

> Producing a Transparent Statement is also referred to as a form of notarization.

Reviewer 2: ```suggestion
```

Author: I've removed line 123 above: 
> Producing a Transparent Statement is also referred to as a form of notarization.

Added:

> Producing a Transparent Statement is also referred to as a form of notarization.

: an entity that creates Signed Statements about software Artifacts in the supply chain.
An Issuer may be the owner or author of software Artifacts, or an independent third party such as a reviewer or an endorser.

Ledger (was Registry):

: the verifiable append-only data structure that stores Signed Statements in a Transparency Service.

> SCITT supports multiple Ledger and Receipt formats to accommodate different Transparency Service implementations, such as historical Merkle Trees and sparse Merkle Trees.

Reviewer: ```suggestion
```

Reviewer: I think this is ledger specific implementation detail, not "cross ledger" or "SCITT".

Author: @OR13, while I agree, I'm not sure what you're suggesting we'd change here. I was reading this as how SCITT could operate, acknowledging different implemetnations are still consistent with a SCITT spec.

All messages are sent as HTTP GET or POST requests.

If the Transparency Service cannot process a client's request, it MUST return an HTTP 4xx or 5xx status code, and the body SHOULD be a JSON problem details object ({{RFC7807}}) containing:

- type: A URI reference identifying the problem.
To facilitate automated response to errors, this document defines a set of standard tokens for use in the type field within the URN namespace of: "urn:ietf:params:scitt:error:".

- detail: A human-readable string describing the error that prevented the Transparency Service from processing the request, ideally with sufficient detail to enable the error to be rectified.

> Error responses SHOULD be sent with the `Content-Type: application/problem+json` HTTP header.

Reviewer: why not `application/json`....

Reviewer 3: Out of scope for this PR I would say. Maybe open an issue to discuss it?

Reviewer: yeah, def out of scope, was just wondering if this had been discussed, I will file an issue.

Reviewer: https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/issues/62
